File: knav-fe/src/app/model/Broker.ts
import {TopicMetric} from './TopicMetric';

export class Broker {
  public id: string;
  public host: string;
  public port: number;
  public jmxPort: number;
  public topicMetricList: Array<TopicMetric>;
  public jmxDomains: Array<String>;
  public os: {
    attributes: {
      SystemCpuLoad: number,
      AvailableProcessors: number,
      Arch: string,
      ProcessCpuLoad: number,
      Name: string
    }
  };
  public kafkaVersion: string;
}

File: knav-be/src/main/java/com/codespair/kafka/navigator/kafkanavigatorbe/model/Broker.java
package com.codespair.kafka.navigator.kafkanavigatorbe.model;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;

import java.util.List;

@Builder
@Getter
@EqualsAndHashCode
public class Broker {
  private Integer id;
  private List<TopicMetric> topicMetricList;
  private List<String> jmxDomains;
  private String host;
  private Integer port;
  private Integer jmxPort;
  private OperatingSystem os;
  private String kafkaVersion;
}
